Automattic/jetpack
# Calypsoify: Requiring the Calypsoify package in the Jetpack and Mu-wpcom plugins

Fixes https://github.com/Automattic/vulcan/issues/320

## Proposed changes:

* This PR requires the newly created Calypsoify package in both the Jetpack plugin and the mu-wpcom plugin (but makes sure the class initialization in the mu-wpcom package only happens if a site is Atomic, as that mirrors the current behaviour on Simple sites (the Jetpack module assets are not loaded there)).
* It also modifies the webpack.config.js file (simplified - JS dealt with by the main Jetpack config and in terms of CSS it's just compiling CSS from SCSS), imports the scss file in the js file (in 'src'), and changes how assets are registered by using the Assets package.
* There have been some CSS and JS tweaks to fix things that have broken at some stage in the past - predominantly the option to toggle off the full-screen editor is now hidden if Calypsoify is enabled.
* What has been fixed:
  * Styles related to hiding the full-screen toggle option (via Options menu in post editor) if Calypsoified
* What has been removed:
  * Styles to hide full-screen option from site editor view modes (option no longer exists)
  * Script to force full-screen for iframed site editor (can no longer be anything but that)
  * The scripts relevant to the close button have been removed as it's only relevant for Jetpack sites, and we're removing Calypsoify from self-hosted Jetpack (Atomic sites have a nav-bar in place thanks to the ETK plugin, which overrides that button).
  * The scripts relevant to the revision.php page have been removed as that was only relevant for Simple sites. Instead a change was made directly to admin-plugins to make sure the `mods-gutenberg.js` file there is loading correctly on the revisions page. Diff here: D149239-code
* Additional info:
  * The `calypsoifyGutenberg` variable defined in the main php file includes `closeUrl`, `manageReusableBlocksUrl` and `createNewPostUrl`. `calypsoifyGutenberg` is still needed for the posts navigation on Atomic sites - this is the `closeUrl` in particular, and it is added to the sidebar via the ETK plugin. Removing it here does affect the relevant link on Atomic sites.  The other two variables - `manageReusableBlocksUrl` and `createNewPostUrl'` - appear to be used in `/widgets.wp.com/wpcom-block-editor/calypso.editor.js`. I'm not familiar with this and digging in would take considerably more time and is already out of the scope of this project, so for the sake of not breaking anything else I've left the `calypsoifyGutenberg` variable and associated functions intact.
  * Why don't we get rid of the `admin-plugins` version of Calypsoify? Some of the code is WPcom specific (some in the JS file in admin-plugins). There are also small differences in the main php file so workarounds would need to be added in the mu-wpcom package version if possible. In short we could, but for now I'm opting to keep things as is on Simple. Should there ever be more work done on Calypsoify in the future maybe it would be worthwhile. There are also potential unknowns at this stage and more testing needed if removing the admin-plugins version.

Related diff: D149239-code

### Other information:


## Jetpack product discussion

https://github.com/Automattic/vulcan/issues/320

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

Jetpack self hosted and Atomic: 


* To test changes on a self hosted site and Atomic, apply this PR.
* If applying the PR locally, make sure to build the assets with `jetpack build packages/calypsoify --production`, jetpack build packages/jetpack-mu-wpcom  --production` and `jetpack build plugins/mu-wpcom-plugin  --production`. 
* If using the Jetpack Beta plugin to test, follow the steps in the generated comment below, remember to apply the PR both for the Jetpack plugin and the 'WordPress.com Features' plugin via the Beta tester UI. More info here too on WoA testing: PCYsg-Osp-p2 
* Visit the wp-admin post editor for a post (not using the Classic Editor), and add the `calypsoify=1` param to the URL. Example: `https://example.com/wp-admin/post.php?post=827&action=edit&calypsoify=1`.
* Confirm the Calypso stylesheet / scripts are loaded when the query param is added by checking the page source (see `calypsoify_wpadminmods_js-css` and `calypsoify_wpadminmods_js-js-extra`).
* Click on the 'options' menu, confirm that you see no option to toggle Fullscreen mode off.
* Go back to the post and remove the `&calypsoify=1` from the URL, and refresh. Click on the 'options' menu, and you should see the option to toggle off Fullscreen mode. Click this, and refresh the page to confirm the change sticks.
* Then, Calypsoify the URL and refresh the page again - the editor should now be full-screen, and the option to toggle off full-screen gone from the options menu.
* Open up a revision for the post, and Calypsoify the URL ( for example `/wp-admin/revision.php?revision=122&calypsoify=1`). The revision page should now be full-screen (vs. not without Calypsoification).

Atomic-specific:

- Additionally, on an Atomic test site, Calypsoify a post URL and then click on the site icon (top left). A sidebar should display (this is thanks to the ETK plugin). Hover over 'All posts' and the URL should the WordPress.com posts URL. 
- Do the same but without a Calypsoified post - the URL should be the wp-admin posts URL.

Simple site:

The changes added here should not affect Simple sites. The main thing to check is whether or not additional stylesheets or scripts are being loaded on Simple sites (using the steps to test this PR on Simple from the generated comment below) - checking the page source when Calypsoifying a (non-Classic-editor) post you shouldn't see duplications and assets should be coming from `admin-plugins`.

Note that more realistic way to test Calypsoification is to visit `https://wordpress.com/block-editor/post/` and select a self hosted site.

Fixes related to the revisions page and CSS on Simple (copying the changes made here which also is relevant on Simple)  can be tested here: D149239-code
